sdk/core/core-rest-pipeline/src/httpHeaders.ts

```diff
@@ -57,10 +68,14 @@ class HttpHeadersImpl implements HttpHeaders {
   /**
    * Get the JSON object representation of this HTTP header collection.
    */
-  public toJSON(): RawHttpHeaders {
+  public toJSON(options: { preserveCase?: boolean } = {}): RawHttpHeaders {
     const result: RawHttpHeaders = {};
-    for (const [key, value] of this._headersMap) {
-      result[key] = value;
+    for (const [normalizedName, entry] of this._headersMap) {
+      if (options.preserveCase) {
+        result[entry.name] = entry.value;
+      } else {
+        result[normalizedName] = entry.value;
+      }
```

Comment:
NIT: perhaps factor out the `preserveCase` check out the loop so it fires only once?

Reply:
Hm yeah I'm not sure if the runtime will optimize this correctly, so I'm going to do as you suggest